### PR TITLE
Add Create and Get s3 metadata functionality

### PR DESCRIPTION
**Pull request by tenyo**
This PR adds support for creating/getting a metadata object in S3.

### CODE REVIEW

**Comment by damnski** [comment]
:+1:

File: s3metadatarepository/s3metadatarepository.go [hunk 1]
```diff
@@ -144,25 +153,111 @@ func WithPrefix(prefix string) S3RepositoryOption {
...
+	}
+	key = key + id
+
+	out, err := s.S3.GetObjectWithContext(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(s.Bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		return nil, ErrCode("failed to get metadata object from s3 "+key, err)
+	}
```
Comment (fishnix) [inline s3metadatarepository/s3metadatarepository.go]: `defer out.Body.Close()`

File: s3metadatarepository/s3metadatarepository_test.go [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+		Description:         "The tiniest dataset of mediocre stuff",
+		CreatedAt:           &testTime,
+		CreatedBy:           "Yo Daddy",
+		DataClassifications: []string{},
+		DataFormat:          "file",
+		DataStorage:         "s3",
+		Derivative:          true,
+		DuaURL:              &url.URL{},
+		ModifiedAt:          &testTime,
+		ModifiedBy:          "Yo Mama",
```
Comment (fishnix) [inline s3metadatarepository/s3metadatarepository_test.go]: 😱

File: s3metadatarepository/s3metadatarepository.go (hunk 1 above)
Comment (tenyo) [reply s3metadatarepository/s3metadatarepository.go]: Good call!
It's not in the API examples, but they recommend closing the Body in this doc: https://github.com/aws/aws-sdk-go#making-api-requests

**Pull request merged**